src/EcommerceStatsBundle/Resources/views/snippets-cms/ecommerceStats/module/barchart-body.html.twig

```diff
@@ -0,0 +1,139 @@
+                            {% set subHeight = subHeight | merge({(subGroupColumn): null}) %}
+                        {% endif %}
+
+                        {% if oldSubHeight[subGroupColumn] is not defined %}
+                            {% set oldSubHeight = oldSubHeight | merge({(subGroupColumn): 0}) %}
+                        {% endif %}
+
+                        {% if barHTML[subGroupColumn] is not defined %}
+                            {% set barHTML = barHTML | merge({(subGroupColumn): ''}) %}
+                        {% endif %}
+
+                        {% if colors[colorCount] is not defined %}
+                            {% set colorCount = 0 %}
+                        {% endif %}
+
+                        {% set value = subGroup.getTotals(name) %}
+
+                        {% if not subHeight[subGroupColumn] == null %}
+                            {% set oldSubHeight = oldSubHeight | merge({(subGroupColumn): subHeight[subGroupColumn]}) %}
+                        {% endif %}
+
+                        {% set subHeight = subHeight | merge({(subGroupColumn): barHeight / maxVal * value}) %}
+                        {% set subBarData = {(subGroupName): {className: subGroupColumn, bottom: oldSubHeight[subGroupColumn], backgroundColor: colors[colorCount], width: subBarWidth, height: subHeight[subGroupColumn]}} %}
+                        {% set barHTML = barHTML | merge({(subGroupColumn): (barHTML[subGroupColumn] ?: {}) | merge(subBarData) }) %}
+
+                        {% set color = colors[colorCount] %}
+                        {% set colorSubcategory = colorSubcategory | merge( {(subGroupColumn): (colorSubcategory[subGroupColumn] ?: {}) | merge({(color): subGroup.groupName}) }) %}
+
+                        {% set colorCount = colorCount + 1 %}
+                    {% endfor %}
+
+                    {% set subBarCount = 1 %}
+
+                    {% for html in barHTML %}
+                        {% if (subBarCount == 1) %}
+                            {% set margin = totalBarWidth %}
+                        {% else %}
+                            {% set margin = subBarWidth %}
+                        {% endif %}
+                        {% if html %}
+                            <div class="bar-wrapper" style="height: {{ barHeight }}px; margin-left: {{ margin }}px;">
+                            {% for barData in html %}
+                                <div class="bar {{ barData.className }}" style="bottom: {{ barData.bottom }}px; background-color: {{ barData.backgroundColor }}; width: {{ barData.width }}px; height: {{ barData.height }}px;"></div>
+                            {% endfor %}
+                            </div>
+                        {% endif %}
+                        {% set subBarCount = subBarCount + 1 %}
+                    {% endfor %}
+                {% endif %}
+
+            </div>  {# of .columnBars #}
+
+            <div class="cleardiv"></div>
```

Comment:
cleardivs are still needed? (very old)

Reply:
The graphs here are actually build by floating bars next to each other - so at the moment the cleardiv is actually required, yes. I did try to refactor this template but without completely replacing how charts are being rendered  I didn't have any success in that.